File: Backend/src/Post/FishUp.Post.Models/Responses/Queries/PostDetails.cs
using FishUp.Dispatchers;
using FishUp.Models.Types;

namespace FishUp.Post.Models.Responses.Queries
{
    public class PostDetails : IQueryResponse
    {
        public Guid Id { get; set; }
        public string Author { get; set; }
        public DateTime CreatedDate { get; set; }
        public string Content { get; set; }
        public IEnumerable<AddedComment> Comments { get; set; }
        public IEnumerable<Photo> Photos { get; set; }
    }
}

File: Frontend/src/FishUp/src/models/post-details.ts
import { DateTime } from "ionic-angular";
import { AddedComment } from "./added-comment";

export class PostDetails {
  id: string;
  author: string;
  authorId: string;
  createdDate: DateTime;
  content: string;
  comments : AddedComment[];
  likersIds : string[];
  disLikersIds : string[];
  likesCount: number;
  disLikesCount: number;
}
